### ISSUE TITLE
remove dependency on SSH when bootstrap cluster

### ISSUE THREAD

**Issue by figo** [kubernetes-sigs/cluster-api-provider-vsphere#87]
/kind feature

**issue**
the current problem we encountered while build e2e CI with [VMC](https://cloud.vmware.com/vmc-aws?utm_source=cpc&utm_medium=ot&utm_campaign=cpc-ot-all-t-119-g-en_us-awargoog--&utm_content=awargoog&kxconfid=sm4lyyz1v&kxcampaignid=926678632&creative=301746484520&keyword=%2Baws%20%2Bcloud&matchtype=b&network=g&device=c&k_clickid=85762a1a-e9a6-4b30-9c1e-179adf5430ec) environment, the VMC only has vCenter be external facing, the VMs are not external facing.

the current clusterapi implementation requires clusterctl has direct SSH connection to `target cluster` to extract kubeconfig info, and getting the target cluster status. 

the code path that depends on SSH to target VM is:
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/pkg/cloud/vsphere/utils/utils.go#L162

**Describe the solution you'd like**
clusterapi expecting each provider implement Reconcile() at cluster actuator, 
the code is running at `vsphere-machine-controller` in a container (again, the container is running on the bootstrap cluster), to get target cluster status, there are multiple ways to
extract kubeconfig information without SSH:
1) kube config can be generated if CA is determined.
2) leverage vmtool interface to communicate the configure information out. 

```
                      
[bootstrap cluster] =(SSH/others)=> [target cluster]
```

**Anything else you would like to add:**

the benefit of not depends on SSH to target cluster:
it will not limit us to ensure bootstrap cluster has direct access to target cluster.

/cc @sflxn @sidharthsurana @frapposelli

**Comment by moshloop** [comment]
I think we still need to solve for the remote connectivity, as I am pretty sure machine controller will need to start implementing machine health / readiness checks at some point?

**Comment by akutz** [comment]
Hi @moshloop,

This has been addressed in-part with #267 and will be finished with a PR I intend to open soon. We're aligning CAPV with CAPA, and part of that is the bootstrap process. Since the CAs are either defined as part of the ClusterConfig or generated in the Cluster actuator, the `kubeconfig` can now be generated in Go code in the Cluster `reconcile` method by using the CA to generate a new cert with the `system:masters` in it.

I will link the additional issues/PRs when I open them.
